### PR TITLE
formulae: add licenses for various formulae

### PR DESCRIPTION
**Pull request by waldyrious**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Adding my contribution to #58225. 🙂 Included in this PR: `wget`, `exiftools`, `less`, `speex`, `telnet(d)` and `utf8proc`. Please see the individual commit messages for details on how these expressions were derived.

(As a side note, I gained a new appreciation for clear licensing terms. These licenses have been quite cumbersome to determine, as the extended commit messages attest.)

Also, `brew audit --strict` doesn't seem to recognize SPDX expressions. I hope that's OK, as the discussion in #59582 seemed to indicate a general agreement to support them.

### CODE REVIEW

File: Formula/utf8proc.rb
Comment (Rylan12) [inline Formula/utf8proc.rb:6]:
```suggestion
  license all_of: ["MIT", "Unicode-DFS-2015"]
```

I think you're right about `Unicode-DFS-2015` but I want to get a second opinion. The license listed in the [`LICENSE.md`](https://github.com/JuliaStrings/utf8proc/blob/master/LICENSE.md#unicode-data-license) links to [this license](http://www.unicode.org/copyright.html) which matches [`Unicode-DFS-2015`](https://spdx.org/licenses/Unicode-DFS-2015.html) but these first few lines are missing from the `LICENSE.md` file:

```
Unicode Data Files include all data files under the directories http://www.unicode.org/Public/, http://www.unicode.org/reports/, and http://www.unicode.org/cldr/data/. Unicode Data Files do not include PDF online code charts under the directory http://www.unicode.org/Public/. Software includes any source code published in the Unicode Standard or under the directories http://www.unicode.org/Public/, http://www.unicode.org/reports/, and http://www.unicode.org/cldr/data/.

NOTICE TO USER: Carefully read the following legal agreement. BY DOWNLOADING, INSTALLING, COPYING OR OTHERWISE USING UNICODE INC.'S DATA FILES ("DATA FILES"), AND/OR SOFTWARE ("SOFTWARE"), YOU UNEQUIVOCALLY ACCEPT, AND AGREE TO BE BOUND BY, ALL OF THE TERMS AND CONDITIONS OF THIS AGREEMENT. IF YOU DO NOT AGREE, DO NOT DOWNLOAD, INSTALL, COPY, DISTRIBUTE OR USE THE DATA FILES OR SOFTWARE.
```

**Review by Rylan12 — CHANGES_REQUESTED** [review]
Fantastic work, @waldyrious!

We have added support for SPDX expressions but we decided to go about it in a more "ruby" way that uses hashes and arrays to indicate the expression logic instead of a parsing a string. I added suggestions with the correct style (I think it should be pretty straight forward).

Also, we prefer one commit per formula. You had a few commits that updated two (related) formulae, so it would be great if you could separate those.

Just so you know, we need to wait for a new Homebrew release before we can start using the updated license syntax in formulae, so I'm going to mark this as `do not merge` until then. Once  we have a new release and everything's been adressed we should be good to go!

File: Formula/less.rb
Comment (gromgit) [reply Formula/less.rb:6]: IANAL, but:
1. the phrase "Less License" at the top may actually indicate that "this is the license that applies to the Less software product", rather than "this series of words shall be distinguished from all other identical series of words as the *Less License*"
1. in a legal context, the actual words of the license matter far more than the name attached to said words, so the [duck test](https://en.wikipedia.org/wiki/Duck_test) should apply here...especially if the author's license summary and the actual license text say different things.

Therefore, I think `BSD-2-Clause` is perfectly fine.

File: Formula/telnet.rb
Comment (gromgit) [reply Formula/telnet.rb:6]: That doesn't seem to match anything in the SPDX database. About all that can be definitively said is "this version of telnet is copyrighted by MIT".

File: Formula/utf8proc.rb
Comment (gromgit) [reply Formula/utf8proc.rb:6]: IANAL, but...

The first paragraph specifies a non-exclusive list of data file locations that don't apply in this case, so I'm not surprised the Unicode folks struck it out.

As for the second paragraph, it's a warning that "use implies acceptance", but since that's already a given in the software licensing world, the Unicode folks may have removed it as being unnecessary.

In any case, the terms of the license (which form the legal "meat") fall below the copyright notice. As long as they match `Unicode-DFS-2015`, I think it's safe to claim it is such.

File: Formula/less.rb
Comment (waldyrious) [reply Formula/less.rb:6]: Well, to be perfectly accurate, the license in [less's LICENSE file](https://github.com/gwsw/less/blob/master/LICENSE) is not 100% identical to the [BSD-2-Clause](https://spdx.org/licenses/BSD-2-Clause.html). In the second clause, it changes from

> 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution. 

to simply

> 2. Redistributions in binary form must reproduce the above copyright notice in the documentation and/or other materials provided with the distribution.

Notice how "above copyright notice, this list of conditions and the following disclaimer" was changed to only "the above copyright notice".

None of the SPDX-encoded BSD licenses seem to match this phrasing exactly, but indeed the standard BSD-2-Clause is the closest match. Not sure if that slight deviation means we should treat it as a different license.

File: Formula/telnet.rb
Comment (waldyrious) [reply Formula/telnet.rb:6]: Good catch. The "Export of this software..." paragraph isn't present in any of the MIT variants encoded by SPDX, but after the "WITHIN THAT CONSTRAINT", the text does match the main body of the [MIT-CMU](https://spdx.org/licenses/MIT-CMU.html), in particular due to the "...and that the name of <author> not be used in advertising or publicity pertaining to distribution of the software without specific, written prior permission."

That said, the disclaimer at the end is also different from `MIT-CMU`'s disclaimer. The Fedora wiki contains a more comprehensive [catalog of MIT variants](https://fedoraproject.org/wiki/Licensing:MIT?rd=Licensing/MIT), and the telnet vs. SPDX disclaimer styles appear to match what they call the [Old Style (no advertising without permission)](https://fedoraproject.org/wiki/Licensing:MIT?rd=Licensing/MIT#Old_Style_.28no_advertising_without_permission.29) and the [Old Style with legal disclaimer](https://fedoraproject.org/wiki/Licensing:MIT?rd=Licensing/MIT#Old_Style_with_legal_disclaimer), respectively.

Not sure what we should do about this.

File: Formula/telnetd.rb
Comment (waldyrious) [reply Formula/telnetd.rb:6]: Sure, will do this once we've resolved what to do with the other in-progress discussions.

File: Formula/perl.rb
Comment (waldyrious) [reply Formula/perl.rb:6]: Sure — will do this once we've resolved what to do with the other in-progress discussions.

File: Formula/telnet.rb
Comment (Rylan12) [reply Formula/telnet.rb:6]: We should ask upstream for clarification on this. I don't see any contact information on the homepage, so I'm not sure who to ask. Would it make any sense to the maintainers listed on the Debian site? If it's a different codebase probably not, right?

**Comment by waldyrious** [comment]
@Rylan12 in the interests of making reviewing of this PR easier, I've applied the changes that have been agreed in the inline discussions, and marked them as resolved. For convenience, the changes were:

- Changing SPDX expressions to the license DSL using `any_of: [...]`, `all_of: [...]`
- Removing BSD-2-Clause from `less.rb`, since it differs in one of the clauses from the actual Less License
- Splitting the changes to `exiftool.rb` and `perl.rb` into separate commits

I'll stand by for instructions on how to proceed regarding `telnet.rb`.

File: Formula/telnet.rb
Comment (waldyrious) [reply Formula/telnet.rb]: It seems to me that a good way to resolve this without loss of information would be to submit a PR to SPDX adding the [Old Style (no advertising without permission)](https://fedoraproject.org/wiki/Licensing:MIT?rd=Licensing/MIT#Old_Style_.28no_advertising_without_permission.29) MIT variant listed in the Fedora wiki. That way we could encode the license as 

```ruby
license all_of: ["BSD-4-Clause-UC", "APSL-1.0", "MIT-no-advertising"]
```

(or whatever SPDX identifier it ends up having.)

File: Formula/telnet.rb
Comment (Rylan12) [inline Formula/telnet.rb:6]:
```suggestion
  license all_of: ["BSD-4-Clause-UC", "APSL-1.0"]
```

Let's just get rid of the `MIT_CMU` license for now. We know that it isn't correct.

File: Formula/telnet.rb
Comment (Rylan12) [reply Formula/telnet.rb]: Yeah, I think that's best. I did reach out to the maintainers but the only response I got was from someone who didn't know.

**Review by Rylan12 — CHANGES_REQUESTED** [review]
Looks good, but I think we should still remove the `MIT-CMU` license from `telnet` as its definitely not `MIT-CMU`.

Also, can you separate the `telnet` and `telnetd` changes into two separate commits?

File: Formula/telnet.rb
Comment (Rylan12) [reply Formula/telnet.rb]: Yeah, I think that's best. I did reach out to the maintainers but the only response I got was from someone who didn't know.

File: Formula/telnet.rb
Comment (waldyrious) [reply Formula/telnet.rb:6]: Done!

**Comment by waldyrious** [comment]
@Rylan12 all done — thanks for the quick review! Please take another look when you can. I believe this may now be ready to merge :)

**Review by Rylan12 — APPROVED** [review]

**Comment by Rylan12** [comment]
Great work! Thanks, @waldyrious!

**Comment by waldyrious** [comment]
Thanks for the thorough and helpful reviews, everyone!

**Pull request merged**
